Simplenote/Classes/NoteInfoViewController.swift

```diff
@@ -0,0 +1,157 @@
+//
+//  Created by Kevin LaCoste on 2020-05-15.
+//  Copyright © 2020 Automattic. All rights reserved.
+//
+
+import UIKit
+
+class NoteInfoViewController: UIViewController, UITableViewDataSource {
+    
+    @IBOutlet weak var titleLabel: UILabel!
+    @IBOutlet weak var dismissButton: UIButton!
+    @IBOutlet weak var tablewView: UITableView!
+    
+    @objc var dismissAction: (() -> Void)? = nil
+    @objc var note: Note?
+    @objc var hasSafeAreas = false
+    
+    var cells = [UITableViewCell]()
+    
+    // MARK: - View Lifecycle
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        setUpTitleLabel()
+        setUpTableCells()
+        
+        if dismissAction == nil {
+            dismissButton.isHidden = true
+        }
+        
+        tablewView.dataSource = self
+        tablewView.tableFooterView = UIView()
+        
+        if hasSafeAreas {
+            self.view.frame.size.height -= 36
```

Comment:
Tricky one! Ideally speaking, you'd always want to use Autolayout (and get rid of magic numbers, such as `-36`). In order to nuke this one, there are a couple things that could be done:

1. Pass along the parent's `safeAreaInsets` by means of the `additionalSafeAreaInsets`
  > infoViewController.additionalSafeAreaInsets = self.view.safeAreaInsets;

2. Adjust the layout here, accordingly to the `additionalSafeAreaInsets`

3. Since you're using `SPActionSheet`, you'd probably want to layout the main view, and adjust its size to the intrinsicContentSize. There are few (manual) ways to do this, involving querying the tableView's `contentSize`, and a bit or arithmetic. **But** main goal is always to get rid of inline constants

Reply:
In this case it's a bit tricky as the 36 points are an arbitrary element of the design spec. I'm not sure how to account for them with Auto Layout. So what I've done is reversed the initial logic and removed that padding from the XIB, then only add the padding to the view if the container doesn't have safe area insets applied. The end result is one magic number (as a constant) in only one place instead of two.

Not sure this is the best approach but it's better. Would love to hear your thoughts on how to better tackle this!

Reply:
Couple notes on this one!

- Yes, definitely a tricky problem
- This is actually similar to precalculating TableViewCell's height(s), once, for performance
- [Sample approach here](https://github.com/wordpress-mobile/WordPress-iOS/blob/d0564d3bc8f3d47fbb3e83e1017741f38329429c/WordPress/Classes/ViewRelated/Blog/Blog%20Details/BlogDetailsViewController.m#L343)
- And [this one might also help](https://stackoverflow.com/questions/18746929/using-auto-layout-in-uitableview-for-dynamic-cell-layouts-variable-row-heights)

In broad lines, solving this 100% with Autolayout would probably involve:

1. Defining everything with Autolayout Constraints, usage of [systemLayoutSizeFittingSize](https://developer.apple.com/documentation/uikit/uiview/1622624-systemlayoutsizefittingsize?language=objc) to determine the "minimum required size", 
2. Potentially using [this trick](https://github.com/wordpress-mobile/WordPress-iOS/blob/309e5249dce755650be4ccf8be9e818b237b51e4/WordPress/Classes/ViewRelated/Notifications/Tools/IntrinsicTableView.swift)
3. Forcing the layout system to calculate the minimum required size
4. And finally just doing the frame assignment

(4) is only required because the container itself isn't Autolayout friendly. But with an overhauled SPActionSheet none of this will probably be required, at all.